src/edu/washington/escience/myria/util/DeploymentUtils.java

```diff
@@ -335,24 +349,30 @@ public static void mkdir(final String address, final String remotePath) {
    * @param remotePath e.g. /tmp/test
    */
   public static void rsyncFileToRemote(final String localPath, final String address, final String remotePath) {
-    rsyncFileToRemote(localPath, address, remotePath, false);
+    rsyncFileToRemote(localPath, address, remotePath, null, null);
   }
 
   /**
-   * Copy a local file to a location on a remote machine, using rsync.
+   * Rsync a local deployment to a remote machine.
    * 
    * @param localPath path to the local file that you want to copy from
    * @param address e.g. beijing.cs.washington.edu
    * @param remotePath e.g. /tmp/test
-   * @param useDel if use --del option
+   * @param includes a list of rsync include args
+   * @param excludes list of rsync exclude args
    */
   public static void rsyncFileToRemote(final String localPath, final String address, final String remotePath,
-      final boolean useDel) {
+      final List<String> includes, final List<String> excludes) {
     ArrayList<String> command = new ArrayList<String>();
     command.add("rsync");
     command.add("-rtLDvz");
-    if (useDel) {
-      command.add("--del");
+    /* Remove old files at dest. */
+    command.add("--del");
+    for (String inc : includes) {
```

Comment:
This will certainly cause an NPE if `includes` or `excludes` is `null`. Was this codepath tested? Personally I favor empty collections over `null` (simpler code), so I think you should leave the code as-is and change the callers to supply empty collections in place of `null`.